src/wormhole/_boss.py

```diff
@@ -234,6 +237,12 @@ def rx_welcome(self, welcome):
             # middle of processing the rx_welcome input, and I wasn't sure
             # Automat would handle that correctly.
             self._W.got_welcome(welcome)  # TODO: let this raise WelcomeError?
+
+            permission = welcome.get("permission-required", {})
+            if "hashcash" in permission:
```

Comment:
According to the actualized version of the spec (I know this PR is based on an older proposal), the control flow here is more complex:

- First of all, check whether there is a `permission-required` field at all. If there isn't the rest can indeed be skipped
- If `permission-required` contains `"none"`, skip as well
- Now comes the `hashcash` part
   - To be honest, I'm thinking about prioritizing `hashcash` over `none` in the Rust implementation. Yes, it's more stupid work, but I hope that it makes the transition smoother when the server is under load.
- If none of the above applies, error out (the server only supports permission models we don't know about).